cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java

```diff
@@ -205,7 +208,13 @@ public static HiveSupport checkHiveSupport(@Nullable ClassLoader hiveClassLoader
     }
 
     ClassLoader classLoader = ExploreUtils.getExploreClassloader();
-    return traceExploreDependencies(classLoader, tmpDir);
+    Set<File> additionalJars = new HashSet<>();
+    if (isSparkAvailable()) {
```

Comment:
Should we only ship the Spark assembly jar to the explore container if it is configured to use Spark as the execution engine for Hive? Or is it dynamic so that we always have to ship it?